src/pages/TermsOfService.js

```diff
@@ -1,281 +1,341 @@
+
+You acknowledge that this may require you, in some instances, to
+anonymize sequence data prior to uploading it to the Site. You further
+agree to indemnify and hold harmless Terra of and from any and all
+claims, demands, losses, causes of action, damage, lawsuits, judgments,
+including attorneys' fees and costs, arising out of or relating to your
+uploading, storing, or transferring of PHI without having fully
+implemented the Clinical Features.
+
+Terms as of February 10, 2020.
 `
 
-const styles = {
-  page: {
-    padding: '1rem', minHeight: '100%',
-    display: 'flex', justifyContent: 'center', alignItems: 'center'
-  },
-  box: {
-    backgroundColor: 'rgba(255,255,255,0.85)',
-    borderRadius: 5, width: 800, padding: '2rem', boxShadow: Style.standardShadow
-  }
-}
-
-const termsTitle = div({ style: { color: colors.dark(), fontWeight: 600 } }, [
-  span({ style: { fontSize: 36 } }, ['TERRA ']), span({ style: { fontSize: 24 } }, ['Terms of Service'])
-])
-
-const TOSMarkdown = div({
-  style: { maxHeight: 400, overflowY: 'auto', lineHeight: 1.5, marginTop: '1rem', paddingRight: '1rem' }
-}, [
-  h(MarkdownViewer, {
-    renderers: {
-      heading: (text, level) => {
-        return `<h${level} style="color: ${colors.dark()}; margin-bottom: 0">${text}</h${level}>`
-      },
-      paragraph: text => {
-        return `<p style="margin-top: 0">${text}</p>`
-      }
-    }
-  }, [termsOfService])
-])
-
-export default class TermsOfService extends Component {
-  constructor(props) {
-    super(props)
-    this.state = { busy: false }
-  }
+const TermsOfServicePage = () => {
+  const [busy, setBusy] = useState()
+  const { isSignedIn, acceptedTos } = authStore.get() // can't change while viewing this without causing it to unmount, so doesn't need to subscribe
 
-  async accept() {
+  const accept = async () => {
     try {
-      this.setState({ busy: true })
+      setBusy(true)
       await Ajax().User.acceptTos()
       authStore.update(state => ({ ...state, acceptedTos: true }))
     } catch (error) {
       reportError('Error accepting TOS', error)
-      this.setState({ busy: false })
+      setBusy(false)
     }
   }
 
-  render() {
-    const { busy } = this.state
-    return div({ role: 'main', style: styles.page }, [
-      backgroundLogo,
-      div({ style: styles.box }, [
-        termsTitle,
-        TOSMarkdown,
-        div({ style: { display: 'flex', justifyContent: 'flex-end', marginTop: '2rem' } }, [
-          h(ButtonSecondary, { style: { marginRight: '1rem' }, onClick: signOut }, 'Cancel'),
-          h(ButtonPrimary, { onClick: () => this.accept(), disabled: busy }, ['Accept'])
-        ])
+  return div({ style: { padding: '1rem', minHeight: '100%', display: 'flex', justifyContent: 'center', alignItems: 'center' } }, [
+    backgroundLogo,
+    div({ style: { backgroundColor: 'rgba(255,255,255,0.85)', borderRadius: 5, width: 800, padding: '2rem', boxShadow: Style.standardShadow } }, [
```

Comment:
We removed the opacity on the data import page at Joy's request. In the name of a11y, I suggest we do the same here.